language/tools/move-mv-llvm-compiler/src/stackless/rttydesc.rs

```diff
@@ -211,25 +216,152 @@ fn define_type_info_global_nil(
+    let ll_fld_count = ll_struct_ty.count_struct_element_types();
+    let mut fld_infos = Vec::with_capacity(ll_fld_count);
+    for i in 0..ll_fld_count {
+        let ll_elt_offset = ll_struct_ty.offset_of_element(dl, i);
+        let ll_ety = ll_struct_ty.struct_get_type_at_index(i);
+        debug!(target: "rtty", "\nmember offset: {}\n{}", ll_elt_offset, ll_ety.dump_properties_to_str(dl));
+
+        // If we're into the compiler-generated fields, get the mtype from the llvm field type.
+        let fld_type = if i < fld_count {
+            s_env.get_field_by_offset(i).get_type()
+        } else {
+            ll_prim_type_to_mtype(llcx, ll_ety)
+        };
+
+        // Get the LLVM literal corresponding to `MoveType` literal for this field.
+        let ll_move_type_literal = tydesc_constant(module_cx, &fld_type, type_display_ctx);
+
+        let ll_offset_val = llvm::Constant::int(ll_int64_ty, U256::from(ll_elt_offset as u64));
+        let ll_fld_info_literal = llcx.const_struct(&[ll_move_type_literal, ll_offset_val]);
+        fld_infos.push(ll_fld_info_literal);
+    }
+
+    // Create the field array global and initialize with `StructFieldInfo`s create above:
+    let aval = llcx.const_array(&fld_infos, ll_fld_info_ty);
+    let ll_fld_array = llmod.add_global2(aval.llvm_type(), "s_fld_array");
+    ll_fld_array.set_constant();
+    ll_fld_array.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+    ll_fld_array.set_unnamed_addr();
+    ll_fld_array.set_initializer(aval.as_const());
+
+    // Create the overall `ll_struct_info_ty` runtime descriptor global. This LLVM type
+    // corresponds to `move_native::rt_types::StructFieldInfo`:
+    //   pub struct StructTypeInfo {
+    //     pub field_array_ptr: *const StructFieldInfo,
+    //     pub field_array_len: u64,
+    //     pub size: u64,
+    //     pub alignment: u64,
+    //   }
+    let ll_struct_type_info_ty = llcx.get_anonymous_struct_type(&[
+        llcx.int_type(8).ptr_type(),
+        ll_int64_ty,
+        ll_int64_ty,
+        ll_int64_ty,
+    ]);
+    let ll_struct_type_info = llmod.add_global(ll_struct_type_info_ty, symbol_name);
+    ll_struct_type_info.set_constant();
+    ll_struct_type_info.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+    ll_struct_type_info.set_unnamed_addr();
+
+    // Create the `StructTypeInfo` initializer.
+    let fld_array_len = llvm::Constant::int(ll_int64_ty, U256::from(ll_fld_count as u64));
+    let struct_size = llvm::Constant::int(ll_int64_ty, U256::from(ll_struct_size as u64));
+    let elt_align = llvm::Constant::int(ll_int64_ty, U256::from(ll_struct_align as u64));
+
+    let ll_struct_type_info_literal =
+        llcx.const_struct(&[ll_fld_array.ptr(), fld_array_len, struct_size, elt_align]);
+    ll_struct_type_info.set_initializer(ll_struct_type_info_literal);
+    ll_struct_type_info
+}
+
+fn ll_prim_type_to_mtype(llcx: &llvm::Context, ll_ty: llvm::Type) -> mty::Type {
+    use mty::{PrimitiveType, Type};
```

Comment:
I was thinking for this of a map: llty -> mty, that we fill in while building lltys from mtys.